src/containers/qvector.c

```diff
@@ -687,6 +687,8 @@ void qvector_free(qvector_t *vector) {
     vector->clear(vector);
     Q_MUTEX_DESTROY(vector->qmutex);
 
+    free(vector->data);
```

Comment:
I think we need to check if vector->data is not NULL because max value is allowed to be set to 0 which case vector->data is pointing NULL.